api/query/cache/index/index_filter_test.go

```diff
@@ -627,6 +627,60 @@ func TestBindExecute_TriagedWildcards(t *testing.T) {
 	assert.Equal(t, expectedResult, srs[0])
 }
 
+func TestBindExecute_TestLabel(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	loader := NewMockReportLoader(ctrl)
+	idx, err := NewShardedWPTIndex(loader, testNumShards)
+	assert.Nil(t, err)
+
+	matchingTestName := "/a/b/c"
+	runs := mockTestRuns(loader, idx, []testRunData{
+		testRunData{
+			shared.TestRun{ID: 1},
+			&metrics.TestResultsReport{
+				Results: []*metrics.TestResults{
+					&metrics.TestResults{
+						Test:   matchingTestName,
+						Status: "PASS",
+					},
+					&metrics.TestResults{
+						Test:   "/d/e/f",
+						Status: "FAIL",
+					},
+				},
+			},
+		},
+	})
+	metadata := map[string][]string{
+		"/foo/bar/b.html": {"random"},
+		matchingTestName:  {"interop1", "INTEROP2"},
+		"/d/e/f":          {""},
+	}
+
+	link := query.Or{Args: []query.ConcreteQuery{query.TestLabel{Label: "interop2", Metadata: metadata}}}
```

Comment:
Is the `query.Or` needed here? There's just a single condition inside, right?

How about tests for "and" and "or" with multiple labels? (Also asked earlier, I'm not sure where it would be best to test that, but at one of the layers seems sufficient.)

Reply:
`query.Or` is not needed and I removed here. I will include the `and` and `or` test in the follow-up PR. But yes, this is the place to test that